src/main/java/org/apache/sysml/runtime/controlprogram/context/ExecutionContext.java

```diff
@@ -600,6 +601,9 @@ public void unpinVariables(List<String> varList, boolean[] varsState) {
 	}
 	
 	public void cleanupCacheableData(CacheableData<?> mo) {
+		if (DMLScript.JMLC_MEMORY_STATISTICS)
+			Statistics.removeCPMemObject(System.identityHashCode(mo));
```

Comment:
Please, add the same in the overloaded method in `SparkExecutionContext` such that it also works through the other APIs.